### DISCUSSION TITLE
Using more than one podIdentity value

### DISCUSSION THREAD

**Discussion by vahidmosi** [kedacore/keda#2271] [Q&A / Need Help]
Hi, 
First of all, thanks for this awesome tool. It actually is very useful.
I'm trying to use Keda with Azure Storage Account Queues. I'm wondering if there is any way to set the podIdentity value per scaledObject? 
Currently it seems the value needs to be set when Keda is installed, and only one value is being set. Is there a workaround for this? I need to define multiple ScaledObjects for different queues and I hope I can have a separate podIdentity for accessing each queue.
Thanks for your help in advance.

**Comment by tomkerkhove** [comment]
We are happy to help!

The problem here is that it will be the KEDA workload that will need to check the Azure resources for its metrics. Because of that, it only allows 1 identity given it is 1 app.

What I typically recommend is using a separate identity purely for autoscaling and grant it permissions accordingly, apart from the applications that depend on it.

This is also done in our .NET Core & Service Bus sample on https://github.com/kedacore/sample-dotnet-worker-servicebus-queue/blob/main/pod-identity.md
